### PR TITLE
SDL Remote control baseline

### PR DESCRIPTION
**Pull request by LuxoftAKutsan**
Remote control baseline SAD changes

### CODE REVIEW

**Comment by LuxoftAKutsan** [comment]
@LevchenkoS @pestOO @okozlovlux please review

**Comment by LuxoftAKutsan** [comment]
@LevchenkoS @pestOO please approve

File: docs/Software Architecture Document/Introduction/index.md [hunk 1]
```diff
@@ -20,6 +20,7 @@ Abbreviations used in this document please find in the table below.
 | SEE              | Software Engineering Environment |
 | TTS              | Text To Speech                   |
 | VR               | Voice Recognition                |
+| CAN              | Controller Area Network          |
```
Comment (pestOO) [inline docs/Software Architecture Document/Introduction/index.md]: I don't see any CAN usage abbreviation in the SAD.
Is it redundant term or some more description is not provided?

File: docs/Software Architecture Document/Components View/index.md [hunk 2]
```diff
@@ -229,6 +245,27 @@ The view is represented by module and subsystem diagrams that show the system's
...
+  - *Behavior*
+    - Receives data from CoreService
+    - Parses data
+    - Creates commands.
+    - Handles incoming HMI notifications
+    - Sends RPC to HMI
+    - Sends RPC to mobile
+    - Extends basic applications with additional RPC's
+  - *Constraints*
+    - N/A
```
Comment (pestOO) [inline docs/Software Architecture Document/Components View/index.md]: Please, provide a link to the RC requirements as a part of Constraints.

File: docs/Software Architecture Document/Components View/index.md [hunk 3]
```diff
@@ -148,6 +148,22 @@ The view is represented by module and subsystem diagrams that show the system's
     - Not work for Android apps.
     - Not work for apps connected via SDL protocol version lower than 4.
 
+#### Plugin Manager
+  - *Responsibility*
+    - Loads all .so files from specific directory, checking if they’re exporting required methods
+    - Stores information about plugin capabilities
+    - Check's if plugin's is able to process RPC
```
Comment (pestOO) [inline docs/Software Architecture Document/Components View/index.md]: Are `'` redundant?

Comment (pestOO) [inline docs/Software Architecture Document/Components View/index.md]: Maybe something like `checks plugins capability to handle RPCs`?

File: docs/Software Architecture Document/View-to-View Relations/index.md [hunk 4]
```diff
@@ -21,6 +21,7 @@ The following table is a mapping between the elements in the Component view and
 | ***Connection Handler***  |                       | X                  |                     |              |
 | ***Security Manager***    |                       | X                  |                     |              |
 | ***HMI Message Handler*** |                       |                    | X                   | X            |
+| ***Plugin Manager***      | X                     |                    |                     |              |
```
Comment (pestOO) [inline docs/Software Architecture Document/View-to-View Relations/index.md]: Where is a RC component mapping to the layer?

**Comment by LuxoftAKutsan** [comment]
@pestOO this PR is accumulation of #25 #24 #26 #28 #30 PR's. Most of you findings should be found there.

**Comment by LuxoftAKutsan** [comment]
@AGaliuzov is not related to Open SDL so he should not review this PR.

**Comment by LuxoftAKutsan** [comment]
> Where is RC UseCases in Use Case View?
Changes was moved from Reverse SAD. Probably some new use cases can be added, but current are rather actusl

**Comment by LuxoftAKutsan** [comment]
@pestOO we have requirements in Jira. We didn't have step of arhitecure coverage. Because functionality and arhitecture and requirenments was moved from Release RevSDL

**Comment by LuxoftAKutsan** [comment]
@pestOO 
> Why 'Remote control' component in a Component View RC component located in a Protocol layer components list?

This is fail but it should be found in #25 reviw

**Comment by LuxoftAKutsan** [comment]
@pestOO  
> Do we have any new RC-realted data to be added to Data View?

All SAD changes are according to RevSDL SAD

File: docs/Software Architecture Document/Components View/index.md (hunk 2 above)
Comment (LuxoftAKutsan) [inline docs/Software Architecture Document/Components View/index.md]: @pestOO we have no O-S Rev requirements yet.

File: docs/Software Architecture Document/View-to-View Relations/index.md (hunk 4 above)
Comment (LuxoftAKutsan) [inline docs/Software Architecture Document/View-to-View Relations/index.md]: Plugin manager is RC component

Comment (LuxoftAKutsan) [inline docs/Software Architecture Document/View-to-View Relations/index.md]: @pestOO Plugin manager is RC components. And THeere is not additional layer.

File: docs/Software Architecture Document/Components View/index.md (hunk 2 above)
Comment (pestOO) [inline docs/Software Architecture Document/Components View/index.md]: @LuxoftAKutsan, 
Requirments are the base of Architecture. Architecture could not be constructed and designed  without it.

File: docs/Software Architecture Document/View-to-View Relations/index.md (hunk 4 above)
Comment (pestOO) [inline docs/Software Architecture Document/View-to-View Relations/index.md]: @LuxoftAKutsan, 
But it's still SDL component described in the component view. And it's related to one of the existing layers.

**Comment by LuxoftAKutsan** [comment]
Not actual

**Comment by JackLivio** [comment]
@LuxoftAKutsan Please add new documents to config.json

File: docs/Software Architecture Document/Component Interaction View/assets/plugin_manager_interaction_view.puml [hunk 5]
```diff
@@ -0,0 +1,55 @@
+@startuml
```
Comment (JackLivio) [inline docs/Software Architecture Document/Component Interaction View/assets/plugin_manager_interaction_view.puml]: Where is this file being used?

**Comment by jacobkeeler** [comment]
Closing in favor of https://github.com/smartdevicelink/sdl_core_documentation/pull/1

**Pull request closed without merging**